flake.nix

```diff
@@ -232,11 +227,34 @@
               gotools
               go-tools
               gofumpt
-            ];
+            ]);
             shellHook = ''
               ${pre-commit.shellHook}
             '';
           };
+
+          tf = {
+            inherit (pkgs) terraform opentofu;
+          };
+        in
+        {
+          inherit terraformProviders;
+
+          packages = getPackages tf.opentofu;
+
+          checks = getChecks tf.opentofu;
+
+          lib = {
+            mkDevShell-terraform = mkDevShell tf.terraform;
+            mkDevShell-opentofu = mkDevShell tf.opentofu;
+            mkDevShell = mkDevShell tf.opentofu;
```

Comment:
Since `lib` provides arbitrary Nix values, that can be functions (as opposed to `packages` and `checks` for example), I think we don't need to bother with those `mkDevShell-xxx`. Maybe we can just have `mkDevShell {terraform ? tf.opentofu}: mkDevShell tf.opentofu` or something like that?

Reply:
whoops, thanks, i adjusted this now